usefulUtilities/airAndWhiteboardDraw/entityResources/entityClientScripts/drawSphereClient.js

```diff
@@ -0,0 +1,743 @@
+    var CLOSE_SOUND_VOLUME = 0.02;
+    
+    var MINIMUM_TRIGGER_PRESS_VALUE = 0.97;
+    var DEFAULT_NORMAL = { x: 0, y: 0, z: 1 };
+    var MAX_LINE_POINTS = 100;
+    var DEFAULT_LINE_PROPERTIES = {
+        type: "PolyLine",
+        name: "Whiteboard Polyline",
+        isUVModeStretch: true,
+        lifetime: DECAY_TIME_S,
+        collisionless: true,
+        grab: { grabbable: false }
+    };
+    
+    var tablet = Tablet.getTablet('com.highfidelity.interface.tablet.system');
+
+    var dominantHandJoint;
+    var dominantHand;
+    var parentJointIndex;
+
+    var whiteboard = null;
+    var whiteboardParts = [];
+    var whiteboardProperties;
+
+    var mouseEventsConnected = false;
+    var controllerMappingName = 'Hifi-DrawApp';
+    var controllerMapping;
+
+    var drawInterval = null;
+    var deletingInterval;
+
+
+    var activeTriggerPress = false;
+    var activeGripPress = false;
+    var drawingInDesktop;
+    
+    var animationData = {};
+    var animationHandlerID;
+
+    var injector;
+    
+    var pickRay;
+
+    var polyLine = null;
+    var lineStartPosition;
+    var previousLinePoint;
+    var previousNormal;
+    var previousStrokeWidth;
+    var currentPoint;
+    var currentNormal;
+    var currentStrokeWidth;
+    var wasLastPointOnBoard;
+    var displacementFromStart;
+
+    var laser = null;
+    
+    var readyToDraw = false;
+    var initialLineStartDataReady = false;
+
+    var PaintSphere = function() {
+        _this = this;
+    };
+
+    PaintSphere.prototype = {
+        preload: function(entityID) {
+            _this.entityID = entityID;
+            Script.setTimeout(function() {
+                var properties = Entities.getEntityProperties(_this.entityID, ['userData','color']);
+                _this.color = properties.color;
+                _this.texture = JSON.parse(properties.userData).textureURL;
+                _this.findWhiteboard();
+                readyToDraw = true;
+            }, WAIT_FOR_ENTITIES_TO_LOAD_MS);
+            dominantHand = MyAvatar.getDominantHand();
+            dominantHandJoint = (dominantHand === "right") ? "RightHand" : "LeftHand";
+            MyAvatar.dominantHandChanged.connect(_this.handChanged);
+
+            parentJointIndex = MyAvatar.getJointIndex(dominantHandJoint + "Index4");
+            if (parentJointIndex === -1) {
+                MyAvatar.getJointIndex(dominantHandJoint + "Index3");
+            }
+            if (parentJointIndex === -1) {
+                MyAvatar.getJointIndex(dominantHandJoint);
+                print("ERROR: Falling back to dominant hand joint as index finger tip could not be found");
+            }
+            tablet.tabletShownChanged.connect(_this.tabletShownChanged);
+            HMD.displayModeChanged.connect(_this.displayModeChanged);
+            Window.domainChanged.connect(_this.domainChanged);
+
+            _this.registerControllerMapping();
+            if (HMD.active) {
+                HMD.closeTablet();
```

Comment:
why do you need to force the tablet closed here?

Reply:
I don't anymore...that was left over from the draw app